### PR TITLE
Added additional index proposals for performance optimization based o…

### PR DESCRIPTION
**Pull request by cast**
additional indexes based on access behaviour of JpaSagaStore and JdbcSagaStore in case of high number of saga's and saga associations.

### CODE REVIEW

File: part4/performance-tuning.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ If you have generated the tables automatically using your JPA implementation (e.
 -   Sagas
 
     Table 'AssociationValueEntry', columns `associationKey` and `sagaId`,
+    
+    Table 'AssociationValueEntry', columns `sagaId`,
+    
+    Table 'AssociationValueEntry', columns `sagaType` and `sagaId`,
```
Comment (abuijze) [inline part4/performance-tuning.md]: I believe, by turning these two columns around, that the index on 'sagaId' alone should not be of any added value. Furthermore, it is best to have the most unique column in front of any columns that contains less unique values. Hence, 'sagaId', 'sagaType' would be preferred.

Also, we'd probably need an index on 'associationKey' and 'sagaType' for the "findSagas" functionality.

File: part4/performance-tuning.md [hunk 2]
```diff
@@ -28,6 +28,12 @@ If you have generated the tables automatically using your JPA implementation (e.
...
+    
+    Table 'AssociationValueEntry', columns `associationKey` and `associationValue`
```
Comment (abuijze) [inline part4/performance-tuning.md]: What use case does this index support?

Comment (cast) [inline part4/performance-tuning.md]: It is used to find saga's by association key and value through https://github.com/AxonFramework/AxonFramework/blob/e29ea8b6202ab5f58ce24a2158c084e01664af1a/core/src/main/java/org/axonframework/eventhandling/saga/repository/jpa/JpaSagaStore.java#L216 although I assume that by value in that case should be enough.

File: part4/performance-tuning.md (hunk 1 above)
Comment (cast) [inline part4/performance-tuning.md]: I agree on your first statement.

I would say that the index on the value would be of more benefit due to high cardinality.

**Comment by abuijze** [comment]
Updated 3.1 documentation, taking comments into consideration.

**Pull request closed without merging**